internal/distributed/datacoord/client/client.go

```diff
@@ -458,16 +482,21 @@ func (c *Client) GetSegmentInfo(ctx context.Context, req *datapb.GetSegmentInfoR
 // 	the root reason is each `SaveBinlogPaths` will overwrite the checkpoint position
 //  if the constraint is broken, the checkpoint position will not be monotonically increasing and the integrity will be compromised
 func (c *Client) SaveBinlogPaths(ctx context.Context, req *datapb.SaveBinlogPathsRequest) (*commonpb.Status, error) {
-	// FIXME(dragondriver): why not to recall here?
-	client, err := c.getGrpcClient()
-	if err != nil {
-		return &commonpb.Status{
-			ErrorCode: commonpb.ErrorCode_UnexpectedError,
-			Reason:    err.Error(),
-		}, nil
-	}
 
-	return client.SaveBinlogPaths(ctx, req)
+	ret, err := c.recall(func() (interface{}, error) {
```

Comment:
I wonder why we don't recall `SaveBinlogPaths` before, but this patch does.

Reply:
@XuanYang-cn  please check this.

Reply:
We don't need to recall this, if failed, DN will redo the flush.
Also we need to guarantee the order, if retry this, it'll break the order.

Reply:
You won! @XuanYang-cn